src/main/java/com/zzpj/services/impl/BookServiceImpl.java

```diff
@@ -9,8 +10,29 @@
 @Service
 public class BookServiceImpl extends BaseServiceImpl<BookRepository, Book> implements BookService {
 
+    private final BookRepository bookRepository;
+
     @Autowired
     public BookServiceImpl(BookRepository bookRepository) {
         super(bookRepository);
+        this.bookRepository = bookRepository;
+    }
+
+    @Override
+    public void add(Book book) {
+        book.setVersion(0L);
+        bookRepository.save(book);
+    }
+
+    @Override
+    public void update(Book book) {
+        Book bookFromRepository = bookRepository.findById(book.getId())
+                .orElseThrow(() -> entityNotFoundException(book.getId()));
+        book.setVersion(bookFromRepository.getVersion());
+        bookRepository.save(book);
+    }
+
```

Comment:
to chyba da radę zrobić w BaseService :P

Reply:
Myślę, że da radę zrobić by każdy Model danych dziedziczył po jakimś interfejsie, który by na rozwiązał sprawę i byśmy uniknęli powielenia kodu :P